frame/transaction-payment/src/lib.rs

```diff
@@ -527,6 +538,31 @@ mod tests {
 		});
 	}
 
+	#[test]
+	fn signed_extension_transaction_payment_multiplied_refund_works() {
+		ExtBuilder::default()
+			.balance_factor(10)
+			.base_fee(5)
+			.build()
+			.execute_with(||
+		{
+			let len = 10;
+			NextFeeMultiplier::put(Fixed128::from_rational(1, NonZeroI128::new(2).unwrap()));
+
+			let pre = ChargeTransactionPayment::<Runtime>::from(5 /* tipped */)
+				.pre_dispatch(&2, CALL, &info_from_weight(100), len)
+				.unwrap();
+			assert_eq!(Balances::free_balance(2), 200 - 5 - 15 - 150 - 5);
+
+			assert!(
+				ChargeTransactionPayment::<Runtime>
+					::post_dispatch(pre, &info_from_weight(100), &post_info_from_weight(50), len, &Ok(()))
+					.is_ok()
+			);
+			assert_eq!(Balances::free_balance(2), 200 - 5 - 15 - 75 - 5);
```

Comment:
```suggestion
			// 75 (3/2 of the returned 50 units of weight ) is refunded
			assert_eq!(Balances::free_balance(2), 200 - 5 - 15 - 75 - 5);
```

Reply:
these are just recommendations and I just wrote down how I checked the test. I think they are helpful.